pycloudlib/lxd/cloud.py

```diff
@@ -78,20 +155,60 @@ def init(
             profile_list: list, optional, profile(s) to use
             user_data: used by cloud-init to run custom scripts/configuration
             config_dict: dict, optional, configuration values to pass
+            is_vm: boolean, optional, defines if a virtual machine will
+                   be created
+            use_ssh: boolean, optional, set instance to execute commands
+                     through ssh instead of lxc exec.
 
         Returns:
             The created LXD instance object
 
         """
-        if ':' not in release:
-            release = self._daily_remote + ':' + release
+        self.key_pair = None
+        base_release = image_id
 
+        profile_list = profile_list if profile_list else []
+        config_dict = config_dict if config_dict else {}
+
+        release = self._set_release_image(image_id, is_vm)
         self._log.debug("Full release to launch: '%s'", release)
 
-        cmd = ['lxc', 'init', release]
+        cmd = ['lxc', 'init', release, name]
+
+        if is_vm:
+            cmd.append('--vm')
+
+            if not profile_list:
+                profile_name = "vm-{}".format(base_release)
 
-        if name:
-            cmd.append(name)
+                self.create_profile(
+                    profile_name=profile_name,
+                    profile_config=base_vm_profiles[base_release]
+                )
+
+                profile_list = [profile_name]
+
+        if use_ssh:
+            error_msg = (
```

Comment:
I have updated that part of the code, since I realized my checks here were not making much sense in the context of an user providing its own ssh keys to issue commands.